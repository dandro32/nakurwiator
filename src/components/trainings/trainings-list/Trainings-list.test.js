import React from 'react';
import TestRenderer from 'react-test-renderer';
import TrainingsList from './Trainings-list';

jest.mock('../../UI/section/Section', () => 'SectionLoadingComponent');
jest.mock('../../UI/table/Table-wrapper', () => () => 'table-wrapper');
const mockFetchTrainings = jest.fn();

describe('Training List components', () => {
  let componentInstance;
  let json;
  const mockTrainings = {
    data: [1,2,3],
    isLoading: false,
    isLoaded: true, 
  };
  beforeEach(() => {
    const renderer = TestRenderer.create(
      <TrainingsList fetchTrainings={mockFetchTrainings} trainings={mockTrainings}/>
    );
    componentInstance = renderer.getInstance();
    json = renderer.toJSON();
  });

  it('should render properly component output', () => {
    expect(json).toEqual(3);
  });

});