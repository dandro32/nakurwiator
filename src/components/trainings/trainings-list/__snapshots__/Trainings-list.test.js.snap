// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Training List components should render properly component output 1`] = `
<div>
  <Table wrapper
    columns={
      Array [
        Object {
          "Header": "ID",
          "accessor": "id",
        },
        Object {
          "Header": "Data",
          "accessor": [Function],
          "id": "date",
        },
        Object {
          "Header": "Twoja masa",
          "accessor": [Function],
          "id": "mass",
        },
        Object {
          "Header": "Białeczko",
          "accessor": "protein",
        },
        Object {
          "Header": "Ile cisłeś",
          "accessor": [Function],
          "id": "time",
        },
        Object {
          "Header": "Ćwiczonka",
          "accessor": [Function],
          "id": "exercises",
        },
        Object {
          "Header": "Kaj to robiłeś",
          "accessor": "where",
        },
      ]
    }
    data={
      Array [
        "test",
      ]
    }
    deleteRow={[Function]}
    icon={
      Object {
        "icon": Array [
          640,
          512,
          Array [],
          "f44b",
          "M104 96H56c-13.3 0-24 10.7-24 24v104H8c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h24v104c0 13.3 10.7 24 24 24h48c13.3 0 24-10.7 24-24V120c0-13.3-10.7-24-24-24zm528 128h-24V120c0-13.3-10.7-24-24-24h-48c-13.3 0-24 10.7-24 24v272c0 13.3 10.7 24 24 24h48c13.3 0 24-10.7 24-24V288h24c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8zM456 32h-48c-13.3 0-24 10.7-24 24v168H256V56c0-13.3-10.7-24-24-24h-48c-13.3 0-24 10.7-24 24v400c0 13.3 10.7 24 24 24h48c13.3 0 24-10.7 24-24V288h128v168c0 13.3 10.7 24 24 24h48c13.3 0 24-10.7 24-24V56c0-13.3-10.7-24-24-24z",
        ],
        "iconName": "dumbbell",
        "prefix": "fas",
      }
    }
    isLoaded={true}
    isLoading={false}
    title="Lista ćwiczeń"
  />
</div>
`;
